### ISSUE TITLE
Define When Operand's Order is Important in Leonidas

### ISSUE THREAD

**Issue by dormaayan** [SpartanRefactoring/Spartanizer#949]
- For Commutative infixExpressions ignore the order of operands
- For non-Commutative force specific order as was in the pattern.
-Support both cases of 2 operands and also more than 2 operands

**Comment by orimarco** [comment]
You probably thought about it but note that `&&` and `||` are not commutative although that logically they are...
But sometimes we would like to have a tipper which doesn't care about the order between && expression (for example- `'a' <= c && c <= 'z'`), so if you could still support an option to cancel non-commutativity in some cases that would be great!

**Comment by dormaayan** [comment]
@orimarco :
Yes, after I thought about it yesterday I understood that there are many cases that we need to force the order of the operands, especially in cases that the same variable appear more than one time in the code, for example here : 

` add(TipperFactory.patternTipper("$X == 0 ? 0 : 1", "Bit.of($X)", ""));
`

here that is really important that we will force the right order in order to know what is the value of $X
So, I will force order as a default and will add option to do ignore that in commutative operands.

**Comment by yossigil** [comment]
Note that there is a function "as.bit(x)", we may also want "sign.of(x)", which returns 0,1,-1

**Comment by yossigil** [comment]
@orimarco ? What's up with this?
